File: redact-java/src/main/java/us/jsy/redact/Strategy.java
package us.jsy.redact;

import java.util.Objects;
import java.util.function.Function;

public enum Strategy implements Function<String, String> {
   FULL {
      public String apply(String s) {
         return "***";
      }
   },
   NOOP {
      public String apply(String s) {
         return s;
      }
   },
   HASH {
      public String apply(String s) {
         return Integer.toHexString(Objects.hash(s));
      }
   },
   LAST_4 {
      public String apply(String s) {
         if (s != null && s.length() > 4) {
            return "*" + s.substring(s.length() - 4);
         } else {
            return s;
         }
      }
   };
}

File: redact-kotlin/src/main/java/reuse/redact/Strategy.kt
package reuse.redact

import java.util.*
import java.util.function.Function

enum class Strategy : Function<String, String> {
    FULL {
        override fun apply(s: String): String {
            return "***"
        }
    },
    NOOP {
        override fun apply(s: String): String {
            return s
        }
    },
    HASH {
        override fun apply(s: String): String {
            return Integer.toHexString(Objects.hash(s))
        }
    },
    LAST_4 {
        override fun apply(s: String): String {
            return if (s != null && s.length > 4) {
                "*" + s.substring(s.length - 4)
            } else {
                s
            }
        }
    }
}